stac_fastapi/html/middleware.py

```diff
@@ -182,19 +182,63 @@ async def send_as_html(message: Message):
             ) and not request.query_params.get("f", ""):
                 encode_to_html = True
 
-            if start_message["status"] == 200 and encode_to_html:
+            response_headers = MutableHeaders(scope=start_message)
+
+            # stac-fastapi application overwrite the content-type for
+            # openapi response and use "application/vnd.oai.openapi+json;version=3.0"
+            if (
+                response_headers.get("Content-Type")
+                == "application/vnd.oai.openapi+json;version=3.0"
+            ):
+                openapi_doc = json.loads(body.decode())
+                for _, path in openapi_doc.get("paths").items():
+                    if (
+                        path.get("get", {}).get("summary", "").lower()
```

Comment:
openapi will use camel case for the summary, so by using `lower` case we make sure the name can match the one provided in `endpoints_names`